frontend/packages/console-app/src/redux/reducers/guided-tour-reducer.ts

```diff
@@ -0,0 +1,40 @@
+import { GuidedTourSidebarActions, Actions } from '../actions/guided-tour-actions';
+import { RootState } from '@console/internal/redux';
+import { GuidedTourStatus } from '../../components/guided-tours/utils/guided-tour-status';
+
+export const guidedTourReducerName = 'guidedTour';
+
+type State = {
+  isExpanded: boolean;
+  activeTourID: string;
+};
+
+const initialState: State = {
+  isExpanded: false,
+  activeTourID: '',
+};
+
+export default (state = initialState, action: GuidedTourSidebarActions) => {
+  if (action.type === Actions.ToggleGuidedTourSidebarExpanded) {
+    return {
+      isExpanded:
+        state.isExpanded &&
+        action.payload.activeTourID &&
+        action.payload.activeTourID !== state.activeTourID
+          ? state.isExpanded
+          : !state.isExpanded,
+      activeTourID: action.payload.activeTourID,
```

Comment:
activeTourId is the tour currently in focus in the sidebar, and when the user starts the tour the status is set to `inProgress`.
the activeTour is needed onClick of the the Tour tile to set the content of the sidebar, so i added it into the toggle action.

Reply:
I agree with Sahil here. We need two separate actions to update the state. One will update the expand collapse state and other one will update the status of active tour because there might be a case where the status of the active tour needs to be updated from some other component and then you wouldn't want to also collapse your sidebar.

Reply:
I don't think we need 2 states.

We just need to know which tour should be shown in the space.
- If no tour ID is set, the drawer should be closed. 
- If a tour ID is set, the drawer should be opened.

There won't be a case where a tour ID is set and you the drawer should be expanded. That would show a blank drawer which has no use.